pkg/yamlmeta/walk.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package yamlmeta
+
+// Visitor performs an operation on the given Node while traversing the AST.
+// Typically defines the action taken during a Walk().
+type Visitor interface {
+	visit(Node) error
```

Comment:
is this private intentionally? if so, any reason why not make walk private as well?

Reply:
Yes, I did this intentionally. I didn't necessarily want to force the consuming programmer to export this function.

But one nice feature of an exported function name is that it's obviously the starting point in a set of functions attached to `struct`. So, I might just reverse this.

I see no causal connection between whether yamlmeta.Walk() should be exported and Visitor.visit be...  In fact, in order for this walker to be useful in other packages, it would _need_ to be exported.

Reply:
>  In fact, in order for this walker to be useful in other packages, it would need to be exported.

yeah, that's what i was nudging towards -- nobody outside of this pkg would be able to impl Visitor if visit stays private.

Reply:
+1 this, I stole the walk.go file for the story I'm working on, and struggled to use it without making it Public.